### PR TITLE
Updated Jenkinsfile_CNP for func tests

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-816

### Change description ###
Jenkinsfile updated to handle func tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -14,8 +14,14 @@ def channel = '#bsp-build-notices'
...
 
 def secrets = [
+  's2s-${env}': [
+    secret('microservicekey-bulk-scan-orchestrator', 'S2S_SECRET')
+  ],
   'bulk-scan-${env}': [
-    secret('payments-queue-listen-connection-string', 'PAYMENTS_QUEUE_READ_CONNECTION_STRING')
+    secret('payments-queue-listen-connection-string', 'PAYMENTS_QUEUE_READ_CONNECTION_STRING'),
+    secret('idam-users-bulkscan-username', 'IDAM_USER_NAME'),
+    secret('idam-users-bulkscan-password', 'IDAM_USER_PASSWORD'),
```
Comment (nitinprabhu) [inline Jenkinsfile_CNP]: where are we going to use those secrets in functional tests application yaml or main application yaml ?

File: Jenkinsfile_CNP [hunk 2]
```diff
@@ -14,8 +14,14 @@ def channel = '#bsp-build-notices'
 GradleBuilder builder = new GradleBuilder(this, product)
 
 def secrets = [
+  's2s-${env}': [
+    secret('microservicekey-bulk-scan-orchestrator', 'S2S_SECRET')
```
Comment (nitinprabhu) [inline Jenkinsfile_CNP]: do we need to load it from s2s vault ? Can we not copy secret into bulk scanning vault and load it from there ?

Comment (nitinprabhu) [reply Jenkinsfile_CNP]: Also where are we going to use those secrets in functional tests application yaml or main application yaml

Comment (apereverzin) [reply Jenkinsfile_CNP]: Not sure I can copy the secret from s2s vault - it does not allow me to access it

Comment (apereverzin) [reply Jenkinsfile_CNP]: All these secrets are used in both main and func tests application.yaml

Comment (lgonczar) [reply Jenkinsfile_CNP]: This one is for functional tests only. They create test exception records directly in CCD.

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -29,6 +34,32 @@ static LinkedHashMap<String, Object> secret(String secretName, String envVar) {
 }
 
 withPipeline(type, product, component) {
+  after('akschartsinstall'){
+    // Vars needed for AKS testing
+    env.S2S_NAME = 'bulk_scan_orchestrator'
+    env.S2S_URL = 'http://rpe-service-auth-provider-aat.service.core-compute-aat.internal'
+    env.CORE_CASE_DATA_API_URL = 'http://ccd-data-store-api-aat.service.core-compute-aat.internal'
+    env.PAY_HUB_URL = "http://ccpay-bulkscanning-api-aat.service.core-compute-aat.internal"
```
Comment (lgonczar) [inline Jenkinsfile_CNP]: Do functional tests talk directly to PayHub?

File: Jenkinsfile_CNP (hunk 2 above)
Comment (lgonczar) [reply Jenkinsfile_CNP]: Worth making a TODO to change it to payment processor when possible

Comment (doncem) [reply Jenkinsfile_CNP]: > Can we not copy secret into bulk scanning vault and load it from there ?

all secrets are already copied and ready to use. iirc this was one of the steps/prerequisites for migrating to AKS

Comment (nitinprabhu) [reply Jenkinsfile_CNP]: Please update this as this is incorrect. Refer slack channel for details

File: Jenkinsfile_CNP [hunk 4]
```diff
@@ -29,6 +32,31 @@ static LinkedHashMap<String, Object> secret(String secretName, String envVar) {
 }
 
 withPipeline(type, product, component) {
+  after('akschartsinstall'){
+    // Vars needed for AKS testing
+    env.S2S_NAME = 'bulk_scan_payment_processor'
+    env.S2S_URL = 'http://rpe-service-auth-provider-aat.service.core-compute-aat.internal'
+    env.CORE_CASE_DATA_API_URL = 'http://ccd-data-store-api-aat.service.core-compute-aat.internal'
+    env.IDAM_API_URL = 'https://idam-api.aat.platform.hmcts.net'
+    env.IDAM_CLIENT_REDIRECT_URI = 'https://bulk-scan-orchestrator-aat.service.core-compute-aat.internal/oauth2/callback'
```
Comment (aliveni) [inline Jenkinsfile_CNP]: i think redirect uri should be pointing to `payment processor` not `orchestrator`

**Comment by apereverzin** [comment]
Everything is in PR-111

**Pull request closed without merging**